src/main/scala/chisel3/util/experimental/decode/decoder.scala

```diff
@@ -31,4 +38,38 @@ object decoder extends LazyLogging {
       plaOutput
     }
   }
+
+  /** Use [[EspressoMinimizer]] to generated decoded signals.
+    *
+    * @param input      input signal that contains decode table input
+    * @param truthTable [[TruthTable]] to decode user input.
+    * @return decode table output.
+    */
+  def espresso(input: UInt, truthTable: TruthTable): UInt = apply(EspressoMinimizer, input, truthTable)
+
+  /** Use [[QMCMinimizer]] to generated decoded signals.
+    *
+    * @param input      input signal that contains decode table input
+    * @param truthTable [[TruthTable]] to decode user input.
+    * @return decode table output.
+    */
+  def qmc(input: UInt, truthTable: TruthTable): UInt = apply(QMCMinimizer, input, truthTable)
+
+  /** try to use [[EspressoMinimizer]] to decode `input` by `truthTable`
+    * if `espresso` not exist in your PATH environment it will fall back to [[QMCMinimizer]], and print a warning.
+    *
+    * @param input      input signal that contains decode table input
+    * @param truthTable [[TruthTable]] to decode user input.
+    * @return decode table output.
+    */
+  def apply(input: UInt, truthTable: TruthTable): UInt = try espresso(input, truthTable) catch {
+    case EspressoNotFoundException =>
+      logger.error(
+        """espresso is not found in your PATH, fall back to QMC.
+          |Quine-McCluskey is a NP complete algorithm, may run forever or run out of memory in large decode tables.
+          |To get rid of this warning, please use `decoder.qmc` directly, or add espresso to your PATH.
+          |""".stripMargin
+      )
+      qmc(input, truthTable)
```

Comment:
I sort of screwed up in my previous comments--since this is the generic API and we can always fall back to QMC, we should just catch all `IOExceptions`, but we should forward some of the error message instead of always suppressing and saying "espresso not found".

I think having the `EspressoNotFoundException` is good, but we also need to catch other `java.io.Exceptions`, and report a similar error message but perhaps include some information from the Exception in the logged message.

Reply:
4ecd25f should implement this.